src/Step/Deploy/InstallUpdate/Update/Setup.php

```diff
@@ -7,109 +7,54 @@
-use Psr\Log\LoggerInterface;
+use Magento\MagentoCloud\Util\UpgradeProcess;
+use Magento\MagentoCloud\Step\StepException;
 
 /**
- * Copyright © Magento, Inc. All rights reserved.
- * See COPYING.txt for license details.
+ * @inheritdoc
  */
 class Setup implements StepInterface
 {
-    /**
-     * @var LoggerInterface
-     */
-    private $logger;
-
-    /**
-     * @var Environment
-     */
-    private $environment;
-
-    /**
-     * @var ShellInterface
-     */
-    private $shell;
-
     /**
      * @var FlagManager
      */
-
     private $flagManager;
 
     /**
-     * @var FileList
-     */
-    private $fileList;
-
-    /**
-     * @var DirectoryList
-     */
-    private $directoryList;
-
-    /**
-     * @var DeployInterface
+     * @var UpgradeProcess
      */
-    private $stageConfig;
+    private $upgradeRunner;
 
     /**
-     * @param LoggerInterface $logger
-     * @param Environment $environment
-     * @param ShellInterface $shell
-     * @param DirectoryList $directoryList
-     * @param FileList $fileList
      * @param FlagManager $flagManager
-     * @param DeployInterface $stageConfig
+     * @param UpgradeProcess $upgradeRunner
      */
     public function __construct(
-        LoggerInterface $logger,
-        Environment $environment,
-        ShellInterface $shell,
-        DirectoryList $directoryList,
-        FileList $fileList,
         FlagManager $flagManager,
-        DeployInterface $stageConfig
+        UpgradeProcess $upgradeRunner
     ) {
-        $this->logger = $logger;
-        $this->environment = $environment;
-        $this->shell = $shell;
-        $this->directoryList = $directoryList;
-        $this->fileList = $fileList;
         $this->flagManager = $flagManager;
-        $this->stageConfig = $stageConfig;
+        $this->upgradeRunner = $upgradeRunner;
     }
 
     /**
      * @inheritdoc
+     *
+     * @throws ConfigurationMismatchException
+     * @throws StepException
      */
     public function execute()
     {
         $this->flagManager->delete(FlagManager::FLAG_REGENERATE);
-
         try {
-            $verbosityLevel = $this->stageConfig->get(DeployInterface::VAR_VERBOSE_COMMANDS);
-            $installUpgradeLog = $this->fileList->getInstallUpgradeLog();
-
-            $this->logger->info('Running setup upgrade.');
-
-            $this->shell->execute('echo \'Updating time: \'$(date) | tee -a ' . $installUpgradeLog);
-            $this->shell->execute(sprintf(
-                '/bin/bash -c "set -o pipefail; %s | tee -a %s"',
-                'php ./bin/magento setup:upgrade --keep-generated --ansi --no-interaction ' . $verbosityLevel,
-                $installUpgradeLog
-            ));
-        } catch (\RuntimeException $exception) {
+            $this->upgradeRunner->execute();
+        } catch (\Exception $exception) {
             //Rollback required by database
             throw new StepException($exception->getMessage(), 6, $exception);
```

Comment:
What does it mean error code 6? If you are using custom codes is better to create constant with an appropriate name.

Reply:
@oshmyheliuk 
Thank you very much for the comment.
But could you explain the meaning of this value?
https://github.com/magento/ece-tools/commit/414899313c8cdcc5a27db525e4b8800ebe43b5a1#diff-1c1a326236ff13e56dca5f0910661aa1R83

Reply:
Done

Reply:
Have no idea, I think it was ported from old MCC.